### PR TITLE
Tolerate a null (unset) text substitutions table.

### PR DESCRIPTION
**Pull request by tbitonti**

### CODE REVIEW

**Review by jimblye — APPROVED** [review]

File: org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/SignatureRuleImpl.java [hunk 1]
```diff
@@ -441,14 +441,20 @@ public static boolean matches(Pattern p, CharSequence input) {
 	public Map<String, String> getTextSubstitutions(String inputFileName) {
 		String simpleFileName = FileUtils.getFileNameFromFullyQualifiedFileName(inputFileName);
 
-		Map<String, String> specificUpdates = getSpecificTextUpdates().get(simpleFileName);
+		Map<String, Map<String, String>> specificUpdates = getSpecificTextUpdates();
```
Comment (bjhargrave) [inline org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/SignatureRuleImpl.java:444]: Wouldn't it be better if `getSpecificTextUpdates` never returned null? At least an empty map. Having these null checks is not the best.

File: org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/SignatureRuleImpl.java [hunk 2]
```diff
@@ -441,14 +441,20 @@ public static boolean matches(Pattern p, CharSequence input) {
...
+			Map<String, String> updates = specificUpdates.get(simpleFileName);
+			if (updates != null) {
+				return updates;
+			}
 		}
 
-		for (Map.Entry<Pattern, Map<String, String>> wildcardEntry : getWildCardTextUpdates().entrySet()) {
-			if (matches(wildcardEntry.getKey(), simpleFileName)) {
-				return wildcardEntry.getValue();
+		Map<Pattern, Map<String, String>> wildcardUpdates = getWildCardTextUpdates();
```
Comment (bjhargrave) [inline org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/SignatureRuleImpl.java:452]: Wouldn't it be better if `getWildCardTextUpdates` never returned null? At least an empty map. Having these null checks is not the best.

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/SignatureRuleImpl.java (hunk 2 above)
Comment (tbitonti) [reply org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/SignatureRuleImpl.java:450]: Updated to answer an empty map.

File: org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/SignatureRuleImpl.java (hunk 1 above)
Comment (tbitonti) [reply org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/SignatureRuleImpl.java:444]: Updated to answer an empty map.

**Review by bjhargrave — APPROVED** [review]

**Pull request merged**